Error with cargo install

Hi. 

rustc --version
rustc 1.68.2 (9eb3afe9e 2023-03-27)

When I try to install the app charge I get this error:

`$ cargo install echomap`

> Compiling echomap v0.7.2
> error[E0631]: type mismatch in function arguments
>    --> /home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:146:18
>     |
> 146 |             .map(GeoJson::FeatureCollection)
>     |              --- ^^^^^^^^^^^^^^^^^^^^^^^^^^
>     |              |   |
>     |              |   expected due to this
>     |              |   found signature defined here
>     |              required by a bound introduced by this call
>     |
>     = note: expected function signature `fn(geojson::feature_collection::FeatureCollection) -> _`
>                found function signature `fn(geojson::FeatureCollection) -> _`
> note: required by a bound in `std::iter::Iterator::map`
>    --> /home/jose/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:785:12
>     |
> 785 |         F: FnMut(Self::Item) -> B,
>     |            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::map`
> 
> error[E0599]: the method `flat_map` exists for struct `Map<FilterMap<Map<IntoIter<String>, [closure@main.rs:144:18]>, [closure@main.rs:145:25]>, fn(FeatureCollection) -> GeoJson {FeatureCollection}>`, but its trait bounds were not satisfied
>    --> /home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:147:14
>     |
> 147 |             .flat_map(|g| process_geojson(g, simplification, is_area))
>     |              ^^^^^^^^ method cannot be called due to unsatisfied trait bounds
>     |
>    ::: /home/jose/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/adapters/map.rs:61:1
>     |
> 61  | pub struct Map<I, F> {
>     | -------------------- doesn't satisfy `_: Iterator`
>     |
>     = note: the following trait bounds were not satisfied:
>             `<fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection} as FnOnce<(geojson::feature_collection::FeatureCollection,)>>::Output = _`
>             which is required by `Map<FilterMap<Map<std::vec::IntoIter<std::string::String>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:144:18: 144:21]>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:145:25: 145:28]>, fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection}>: Iterator`
>             `fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection}: FnMut<(geojson::feature_collection::FeatureCollection,)>`
>             which is required by `Map<FilterMap<Map<std::vec::IntoIter<std::string::String>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:144:18: 144:21]>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:145:25: 145:28]>, fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection}>: Iterator`
>             `Map<FilterMap<Map<std::vec::IntoIter<std::string::String>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:144:18: 144:21]>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:145:25: 145:28]>, fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection}>: Iterator`
>             which is required by `&mut Map<FilterMap<Map<std::vec::IntoIter<std::string::String>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:144:18: 144:21]>, [closure@/home/jose/.cargo/registry/src/github.com-1ecc6299db9ec823/echomap-0.7.2/src/main.rs:145:25: 145:28]>, fn(geojson::FeatureCollection) -> GeoJson {FeatureCollection}>: Iterator`
> 
> Some errors have detailed explanations: E0599, E0631.
> For more information about an error, try `rustc --explain E0599`.
> error: could not compile `echomap` due to 2 previous errors
> error: failed to compile `echomap v0.7.2`, intermediate artifacts can be found at `/tmp/cargo-installmIbOn6`


**Cloning the repository, compile correctly**

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,22 +12,22 @@ keywords = ["geo", "map", "gis", "cli"]
 exclude = [".github/*", "img/*", "fixtures/*"]
 
 [dependencies]
-geo = "0.23"
-geo-types = { version = "0.7", features = ["rstar"] }
+geo = "0.24.1"
+geo-types = { version = "0.7.9", features = ["rstar"] }
 geojson = { version = "0.23.0", features = ["geo-types"] }
 num-traits = "0.2"
-rstar = "0.9"
+rstar = "0.10.0"
 anyhow = "1.0"
-console = "0.15"
-indicatif = "0.17"
+console = "0.15.5"
+indicatif = "0.17.3"
 clap = "4.0.22"
 csv = "1.1"
 shapefile = { version = "0.3.0", features = ["geo-types"] }
 topojson = "0.5.1"
 float-cmp = "0.9"
-wkt = "0.10"
-polyline = "0.9"
-kml = "0.4.3"
+wkt = "0.10.3"
+polyline = "0.10.0"
+kml = "0.8.0"
 
 [[bin]]
 name = "echomap"

--- a/src/map_grid.rs
+++ b/src/map_grid.rs
@@ -5,7 +5,7 @@ use anyhow::{Context, Result};
 use geo::algorithm::bounding_rect::BoundingRect;
 use geo::algorithm::contains::Contains;
 use geo::algorithm::intersects::Intersects;
-use geo::algorithm::simplifyvw::SimplifyVW;
+use geo::algorithm::simplify_vw::SimplifyVw;
 use geo::{GeoFloat, Geometry, Line, Point, Polygon, Rect};
 use num_traits::FromPrimitive;
 use rstar::{self, RTree, RTreeNum, RTreeObject, AABB};
@@ -34,21 +34,21 @@ where
             Geometry::MultiPoint(s) => s.into_iter().map(GridGeom::Point).collect(),
             Geometry::Line(s) => vec![GridGeom::Line(s); 1],
             Geometry::LineString(s) => s
-                .simplifyvw(&simplification)
+                .simplify_vw(&simplification)
                 .lines()
                 .map(GridGeom::Line)
                 .collect(),
             Geometry::MultiLineString(s) => s
                 .into_iter()
-                .map(|ls| ls.simplifyvw(&simplification))
+                .map(|ls| ls.simplify_vw(&simplification))
                 .flat_map(|ls| ls.lines().collect::<Vec<_>>())
                 .map(GridGeom::Line)
                 .collect(),
             Geometry::Polygon(s) => {
                 if is_area {
-                    vec![GridGeom::Polygon(s.simplifyvw(&simplification)); 1]
+                    vec![GridGeom::Polygon(s.simplify_vw(&simplification)); 1]
                 } else {
-                    s.simplifyvw(&simplification)
+                    s.simplify_vw(&simplification)
                         .exterior()
                         .lines()
                         .map(GridGeom::Line)
@@ -57,12 +57,12 @@ where
             }
             Geometry::MultiPolygon(s) => {
                 if is_area {
-                    s.simplifyvw(&simplification)
+                    s.simplify_vw(&simplification)
                         .into_iter()
                         .map(GridGeom::Polygon)
                         .collect()
                 } else {
-                    s.simplifyvw(&simplification)
+                    s.simplify_vw(&simplification)
                         .into_iter()
                         .flat_map(|p| p.exterior().lines().collect::<Vec<_>>())
                         .map(GridGeom::Line)

